tests/test_maybe.py

```diff
@@ -193,6 +203,22 @@ def test_slots() -> None:
         n.some_arbitrary_attribute = 1  # type: ignore[attr-defined]
 
 
+def test_some_ok_or() -> None:
+    assert Some(1).ok_or('error') == result.Ok(1)
+
+
+def test_some_ok_or_else() -> None:
+    assert Some(1).ok_or_else('error') == result.Ok(1)
```

Comment:
Yep, I totally missed this precisely because the type checkers didn't flag it.  Fixing the type on the other end indeed causes the type checkers to catch this.